packages/language-server/src/plugins/css/CSSClassDefinitionLocator.ts

```diff
@@ -0,0 +1,167 @@
+        }
+
+        return false;
+    }
+
+    /**
+     * Conditional Expression format:
+     * class="{current === 'baz' ? 'selected' : ''
+     */
+    public IsConditionalExpressionClassFormat() {
+        if (
+            this.initialNodeAt?.type == 'Literal' &&
+            this.initialNodeAt?.parent?.type == 'ConditionalExpression' &&
+            this.initialNodeAt?.parent.parent?.parent?.name == 'class'
+        ) {
+            return true;
+        }
+
+        return false;
+    }
+
+    /**
+     * Class Directive format:
+     * class:active="{current === 'foo'}"
+     */
+    public IsDirectiveFormat() {
+        if (this.initialNodeAt?.type == 'Class' && this.initialNodeAt?.parent?.type == 'Element') {
+            return true;
+        }
+
+        return false;
+    }
+
+    public GetStandardFormatClassName() {
+        const testEndTagRange = Range.create(
+            Position.create(this.position.line, 0),
+            Position.create(this.position.line, this.position.character)
+        );
+        const text = this.document.getText(testEndTagRange);
+
+        let loopLength = text.length;
+        let testPosition = this.position.character;
+        let spaceCount = 0;
+
+        //Go backwards until hitting a " and keep track of how many spaces happened along the way
+        while (loopLength) {
+            const testEndTagRange = Range.create(
+                Position.create(this.position.line, testPosition - 1),
+                Position.create(this.position.line, testPosition)
+            );
+            const text = this.document.getText(testEndTagRange);
+            if (text === ' ') {
+                spaceCount++;
+            }
+
+            if (text === '"') {
+                break;
+            }
+
+            testPosition--;
+            loopLength--;
+        }
+
+        const cssClassName = this.initialNodeAt?.data.split(' ')[spaceCount];
+
+        return this.GetDefinitionRangeWithinStyleSection(`.${cssClassName}`);
+    }
+
+    public GetDefinitionRangeWithinStyleSection(targetClass: string) {
+        let indexOccurence = this.document.content.indexOf(targetClass, 0);
+
+        while (indexOccurence >= 0) {
+            if (this.IsOffsetWithinStyleSection(indexOccurence)) {
+                const startPosition = this.document.positionAt(indexOccurence);
+                const targetRange = Range.create(
+                    startPosition,
+                    Position.create(
+                        startPosition.line,
+                        startPosition.character + targetClass.length
+                    )
+                );
+                indexOccurence = this.document.content.indexOf(targetClass, indexOccurence + 1);
+
+                if (!this.IsExactClassMatch(targetRange)) {
+                    continue;
+                }
+
+                return targetRange;
+            }
+        }
+    }
+
+    public IsOffsetWithinStyleSection(offsetPosition: number) {
```

Comment:
This and other methods should probably be private since they are not used outside of the class.

Reply:
got it